src/MainDriver.cpp

```diff
@@ -850,11 +872,19 @@ int main(Global& glb, const char* souffle_executable) {
 
     // parse file
     ErrorReport errReport(process_warn_opts(glb));
-
     DebugReport debugReport(glb);
-    Own<ast::TranslationUnit> astTranslationUnit = ParserDriver::parseTranslationUnit(
-            glb, InputPath.string(), Input->getInputStream(), errReport, debugReport);
-    Input->endInput();
+
+    std::string sourceBuffer;
+    Input->getInputStream();
+    if (!Input->readInput(sourceBuffer)) {
+        throw std::runtime_error("Failed to read input");
+    }
+    if (!Input->endInput()) {
+        std::exit(1);
```

Comment:
calling `exit` is problematic when we use souffle as a library in a larger program, it would be preferable to either throw an exception or return from the function with an error code.